lib/rubocop/cop/rails/redundant_receiver_in_with_options.rb

```diff
@@ -26,6 +30,31 @@ module Rails
       #       has_many :expenses
       #     end
       #   end
+      #
+      # @example
+      #   # bad
+      #   with_options options: false do |merger|
+      #     merger.invoke(merger.something)
+      #   end
+      #
+      #   # good
+      #   with_options options: false do
+      #     invoke(something)
+      #   end
+      #
+      #   # good
+      #   client = Client.new
+      #   with_options options: false do |merger|
+      #     clinent.invoke(merger.something, something)
+      #   end
+      #
+      #   # maybe good
+      #   with_options options: false do |merger|
+      #     merger.invoke do
+      #       merge.call
```

Comment:
Maybe use something other than `merge` here?  A name/obj that couldn't be confused for the `merger` obj.